src/Groups/matrices/stuff_field_gen.jl

```diff
@@ -0,0 +1,52 @@
+import Hecke: evaluate, field_extension, FinField, FinFieldElem, PolyElem
+
+export primitive_element
+
+function _change_type(f::PolyElem{T}) where T <: FinFieldElem
+   e,p = ispower(order(base_ring(f)))
+   F = GF(Int(p),Int(e))[1]
+   t = PolynomialRing(F,"t")[2]
+   return sum([t^i*F(lift(coeff(f,i))) for i in 0:degree(f)])
+end
+
+# return a generator for the unit group of F
+function _centralizer(f::PolyElem{T}) where T <: FinFieldElem
+  if typeof(f)!=fq_nmod_poly && typeof(f)!=fq_poly
+     f = _change_type(f)
+  end
+  K = base_ring(f)
+  d = degree(K)
+  e = degree(f)
+  q = order(K)
+  L, mL = field_extension(f)
+#  m = divexact(order(L)-1, order(K)-1)
+#  U, mU = unit_group(L, n_quo = Int(m))
+  U, mU = unit_group(L)
+  g = mU(U[1])
+  return mL\g
+end
+
+
+# return a primitive element of F, i.e. a group generator for F*
+# TODO: are there faster procedures?
+"""
+    primitive_element(F::FinField)
+Return a generator of the multiplicative group of `F`.
```

Comment:
So I guess this is something that should ultimately be defined generally in AbstractAlgebra, and implemented for specific finite fields in Nemo.jl and elsewhere, so that we can remove the implementation here. Right?

Reply:
Hmm, I looked through all your PRs but couldn't find any place `primitive_element` is actually used (other than in the tests). Am I overlooking something?

Reply:
It seems you renamed it to `primitive_root` in some PR(s), but not here... argh

Reply:
Yeah, you guess right. The files `matrix_manipulation.jl` and `stuff_field_gen.jl` are made just to furnish me some methods I need in further PR, like `forms` or `jordan-struct`. But those are really row function just to make everything working (a discrete logarithm function as the one implemented in `stuff_field_gen.jl` cannot be exported, it's too bad). The functions here are made to be deleted once the "real" functions are available from other packages. Maybe should I write a couple of lines of comments in these files?

Reply:
Yeah, sorry for the confusion created by the fact that it is renamed `primitive_root` in the `jordan-struct` branch and not here. I did it because Thomas suggested this change to me in the PR on `jordan-struct`. Just leave `primitive_element` here?

Reply:
Why not rename it to `primitive_root` here, too?

Reply:
Claus pointed out that `Hecke.primitive_element` exists -- did you try using it?